sql/src/main/java/io/crate/execution/engine/window/WindowProjector.java

```diff
@@ -39,21 +39,27 @@
     private final List<Input<?>> standaloneInputs;
     private final List<CollectExpression<Row, ?>> standaloneExpressions;
     private final List<WindowFunction> windowFunctions;
+    private final List<? extends CollectExpression<Row, ?>> expressions;
     private final int[] orderByIndexes;
     private final List<DataType> outputTypes;
     private final RamAccountingContext ramAccountingContext;
+    private final Input[][] inputs;
 
     public WindowProjector(WindowDefinition windowDefinition,
                            List<WindowFunction> windowFunctions,
+                           List<? extends CollectExpression<Row, ?>> expressions,
                            List<Input<?>> standaloneInputs,
                            List<CollectExpression<Row, ?>> standaloneExpressions,
                            List<DataType> outputTypes,
                            RamAccountingContext ramAccountingContext,
-                           int[] orderByIndexes) {
+                           int[] orderByIndexes,
+                           Input[]... inputs) {
```

Comment:
Maybe we could clarify `inputs` and `expressions`. We've `standaloneExpressions` and `expressions` - it is unclear what `expressions` relate to. Same with inputs.  (`wfArgsExpressions` / `wfArgsInputs`) maybe?

If we can't come up with a better name I think we'd have to add some docs. Otherwise the constructor is pretty hard to use without digging into the implementation.

Reply:
yeah, good shout, will rename and add some docs